src/main/java/com/exasol/adapter/dialects/exasol/ExasolColumnMetadataReader.java

```diff
@@ -60,15 +58,37 @@ public DataType mapJdbcType(final JDBCTypeDescription jdbcTypeDescription) {
     @Override
     public JDBCTypeDescription readJdbcTypeDescription(final ResultSet remoteColumn) throws SQLException {
         final JDBCTypeDescription typeDescription = super.readJdbcTypeDescription(remoteColumn);
-        if (typeDescription.getJdbcType() == EXASOL_INTERVAL_DAY_TO_SECONDS) {
+        switch (typeDescription.getJdbcType()) {
+        case EXASOL_INTERVAL_DAY_TO_SECONDS:
             return extractIntervalDayToSecondPrecision(remoteColumn, typeDescription);
-        } else if (typeDescription.getJdbcType() == EXASOL_INTERVAL_YEAR_TO_MONTHS) {
+        case EXASOL_INTERVAL_YEAR_TO_MONTHS:
             return extractIntervalYearToMonthPrecision(remoteColumn, typeDescription);
-        } else {
+        case EXASOL_GEOMETRY:
+            return getGeometryWithExtractedSrid(remoteColumn, typeDescription);
+        default:
             return typeDescription;
         }
     }
 
+    private JDBCTypeDescription getGeometryWithExtractedSrid(final ResultSet remoteColumn,
+            final JDBCTypeDescription typeDescription) throws SQLException {
+        final String typeDescriptionString = getTypeDescriptionStringForColumn(remoteColumn);
+        final int srid = extractSrid(typeDescriptionString);
+        return new JDBCTypeDescription(typeDescription.getJdbcType(), typeDescription.getDecimalScale(), srid,
+                typeDescription.getByteSize(), typeDescription.getTypeName());
+    }
+
+    protected int extractSrid(final String typeDescriptionString) {
+        final Pattern pattern = Pattern.compile(SRID_PATTERN);
+        final Matcher matcher = pattern.matcher(typeDescriptionString);
+        if (matcher.find()) {
+            final String srid = matcher.group();
+            return Integer.parseInt(srid.substring(1, srid.length() - 1));
```

Comment:
Use the result of the capture group here.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Matcher.html#group(int)